### PR TITLE
Feature: email token login

### PR DESCRIPTION
**Pull request by offbyone**
- Add a script that sets up dealer membership types
- Add a shortcode login method

This is now complete.

Implements #109

### CODE REVIEW

**Comment by offbyone** [comment]
The more confusing issue can be seen if you follow these steps:
1. Ensure you are logged out on the development instance
2. Log in with an existing account. You will be redirected to the pin entry screen automatically.
3. Enter the pin you got in mailcatcher

You'll see that you are still on the PIN entry screen, _but_ if you are watching the Rails server console, you will _also_ see that your browser make a `GET` request to the reservations URL... but it's not showing it to you. That has me stumped. The controller code for this is in the `enter` method in the `temporary_user_token.rb` controller, and in the corresponding view. I would _love_ to know wtf is happening.

```
Started GET "/user_tokens/enter?shortcode=******&commit=Login%20with%20code" for 127.0.0.1 at 2021-11-21 00:44:16 -0800
Processing by UserTokensController#enter as JS
  Parameters: {"shortcode"=>"d02283", "commit"=>"Login with code"}
Redirected to http://localhost:3000/
Completed 302 Found in 14ms (ActiveRecord: 0.0ms | Allocations: 3057)


Started GET "/" for 127.0.0.1 at 2021-11-21 00:44:16 -0800
Processing by LandingController#index as JS
  Rendering layout layouts/chicago.html.erb
  Rendering landing/index.html.erb within layouts/chicago
  Rendered landing/index.html.erb within layouts/chicago (Duration: 0.0ms | Allocations: 4)
[Webpacker] Everything's up-to-date. Nothing to do
[Webpacker] Everything's up-to-date. Nothing to do
[Webpacker] Everything's up-to-date. Nothing to do
[Webpacker] Everything's up-to-date. Nothing to do
  Rendered layouts/_alerts.html.erb (Duration: 0.1ms | Allocations: 32)
  Rendering user_tokens/new.html.erb
  Rendered user_tokens/new.html.erb (Duration: 2.2ms | Allocations: 781)
  Rendered layouts/_menu.html.erb (Duration: 2.9ms | Allocations: 1114)
  Rendered layout layouts/chicago.html.erb (Duration: 98.0ms | Allocations: 19217)
Completed 200 OK in 99ms (Views: 98.8ms | ActiveRecord: 0.0ms | Allocations: 19712)
```

File: config/webpack/environment.js [hunk 1]
```diff
@@ -15,9 +15,15 @@
...
-const vue = require('./loaders/vue')
+const { VueLoaderPlugin } = require('vue-loader');
+const vue = require('./loaders/vue');
+const webpack = require('webpack');
 
+
+environment.plugins.prepend('jQuery', new webpack.ProvidePlugin({
+  jQuery: 'jquery',
+  $: 'jquery',
+}));
```
Comment (Lasiorhine) [inline config/webpack/environment.js:26]: What kind of blast radius does this change have?

File: spec/commands/token/send_link_spec.rb [hunk 2]
```diff
@@ -36,6 +36,14 @@
...
     end
 
+    it "stores a valid shortcode" do
+      service = Token::SendLink.new(email: good_email, secret: good_secret, path: "")
+      expect(MembershipMailer).to receive_message_chain(:login_link, :deliver_later).and_return(true)
+      expect(service.call).to be_truthy
+      expect(service.errors).to be_none
+      expect(TemporaryUserToken.where(shortcode: service.shortcode)).to be_present
+    end
+
```
Comment (Lasiorhine) [inline spec/commands/token/send_link_spec.rb:46]: Are all the edge and corner cases covered by the existing tests about invalid inputs?

File: spec/commands/token/send_link_spec.rb [hunk 3]
```diff
@@ -48,13 +56,13 @@
       expect(service.call).to be_truthy
     end
 
-    context "with invalid inptus" do
+    context "with invalid inputs" do
       before { expect(MembershipMailer).to_not receive(:login_link) }
```
Comment (Lasiorhine) [inline spec/commands/token/send_link_spec.rb:60]: Yay.

File: app/commands/token/lookup_or_create_user.rb [hunk 4]
```diff
@@ -67,15 +66,11 @@ def errors
   private
 
   def check_token
-    if !token.present?
-      errors << "missing token"
-    end
+    errors << "missing token" unless token.present?
```
Comment (Lasiorhine) [inline app/commands/token/lookup_or_create_user.rb:69]: I like that you tidied all these up.

**Review by Lasiorhine — COMMENTED** [review]
Approved once my questions are answered!

File: config/webpack/environment.js (hunk 1 above)
Comment (offbyone) [reply config/webpack/environment.js:26]: As far as I can tell... just "adds jQuery"

File: spec/commands/token/send_link_spec.rb (hunk 2 above)
Comment (offbyone) [reply spec/commands/token/send_link_spec.rb:46]: I think they are... or at least, this is no worse?
